test/sourcemap.coffee

```diff
@@ -63,3 +64,55 @@ test "#3075: v3 source map fields", ->
   v3SourceMap = JSON.parse v3SourceMap
   arrayEq v3SourceMap.sources, ['tempus_fugit.coffee']
   eq v3SourceMap.sourceRoot, './www_root/coffee/'
+
+# Source maps aren't accurate on Node v12 ??
+if process.version.split(".")[0] != "v12"
+  test "native source maps", ->
+    new Promise (resolve, reject) ->
+      proc = spawn "node", [
+        "--enable-source-maps"
+        "-r", "./register.js"
+        "-r", "./test/integration/error.coffee"
+      ]
+
+      # proc.stdout.setEncoding('utf8')
+      # proc.stdout.on 'data', (s) -> console.log(s)
+      err = ""
+      proc.stderr.setEncoding('utf8')
+      proc.stderr.on 'data', (s) -> err += s
+      proc.on        'exit', (status) ->
+        try
+          equal status, 1
+
+          [_, line] = err.match /error\.coffee:(\d+)/
+          equal line, 3 # Mapped source line
+          resolve()
+        catch e
+          reject(e)
+
+test "don't change stack traces if another library has patched `Error.prepareStackTrace`", ->
+  new Promise (resolve, reject) ->
+    proc = spawn "node", [
+      "-r", "./test/integration/prepare_stack_trace.js",
+      "-r", "./register.js",
+      "-r", "./test/integration/error.coffee",
```

Comment:
I would think that you could test this more directly, without needing to throw an error or create a fixture?

```suggestion
      '--eval', '''
        const patchedPrepareStackTrace = function() {};
        require('./register.js');
        console.log(Error.prepareStackTrace === patchedPrepareStackTrace);
      '''
```

And then check if `stdout` contains `'true'`.